retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs

```diff
@@ -0,0 +1,190 @@
+        }
+        catch (Exception)
+        {
+            Console.WriteLine($"Bucket {bucketName} does not exist.");
+        }
+    }
+
+    /// <summary>Delete all objects from bucket.</summary>
+    public static void DeleteObjectsFromBucket(Bucket bucket)
+    {
+        Console.WriteLine($"Deleting object from bucket {bucket.Name}.");
+
+        var blobs = storageClient.ListObjects(bucket.Name);
+        foreach (var blob in blobs)
+        {
+            storageClient.DeleteObject(bucket.Name, blob.Name);
+        }
+
+        Console.WriteLine($"All objects are deleted from a GCS bucket {bucket.Name}.");
+    }
+
+    /// <summary>Delete all products.</summary>
+    public static void DeleteAllProducts()
+    {
+        Console.WriteLine($"Deleting all products from catalog, please wait.");
+
+        string locationId = "global";
+        string catalogId = "default_catalog";
+        string branchId = "0";
+        BranchName defaultBranch = new BranchName(projectId, locationId, catalogId, branchId);
+
+        var productServiceClient = ProductServiceClient.Create();
+        var listProductsRequest = new ListProductsRequest
+        {
+            ParentAsBranchName = defaultBranch,
+        };
+
+        var products = productServiceClient.ListProducts(listProductsRequest);
+        int deleteCount = 0;
+
+        foreach (var product in products)
+        {
+            var deleteProductRequest = new DeleteProductRequest
+            {
+                Name = product.Name
+            };
+
+            try
+            {
+                productServiceClient.DeleteProduct(deleteProductRequest);
+                deleteCount++;
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Ignore PermissionDenied in case the product does not exist at time of deletion.");
+            }
+        }
+
+        Console.WriteLine($"{deleteCount} products were deleted from {defaultBranch}");
+    }
+
+    /// <summary>Delete Big Query dataset with tables.</summary>
+    public static void DeleteBQDatasetWithData(string datasetId)
+    {
+        Console.WriteLine($"Deleting a {datasetId} BigQuery dataset with all its contents.");
+
+        DatasetReference datasetReference = new DatasetReference
+        {
+            DatasetId = datasetId,
+            ProjectId = projectId
+        };
+
+        DeleteDatasetOptions deleteOptions = new DeleteDatasetOptions
+        {
+            DeleteContents = true
+        };
+
+        try
+        {
+            bigQueryClient.DeleteDataset(datasetReference, deleteOptions);
+
+            Console.WriteLine($"Dataset {datasetId} and contents were deleted.");
+        }
+        catch (Exception)
+        {
+            Console.WriteLine($"Dataset {datasetId} does not exist.");
+        }
+    }
+
+    /// <summary>Delete Big Query dataset with tables.</summary>
+    public static void DeleteBQTable(string datasetId, string tableId)
```

Comment:
If you delete the dataset with its contents, you don't need to delete the tables individually. You can remore this method.

Reply:
done